test/suites/integration/msp/debt-collection.test.ts

```diff
@@ -0,0 +1,250 @@
+    strictEqual(mspNodePeerId.toString(), userApi.shConsts.NODE_INFOS.msp1.expectedPeerId);
+  });
+
+  it("MSP receives files from user after issued storage requests", async () => {
+    const source = ["res/whatsup.jpg", "res/adolphus.jpg", "res/smile.jpg"];
+    const destination = ["test/whatsup.jpg", "test/adolphus.jpg", "test/smile.jpg"];
+    const bucketName = "nothingmuch-3";
+
+    const valueProps = await userApi.call.storageProvidersApi.queryValuePropositionsForMsp(
+      userApi.shConsts.DUMMY_MSP_ID
+    );
+
+    const valuePropId = valueProps[0].id;
+
+    const newBucketEventEvent = await userApi.createBucket(bucketName, valuePropId);
+    const newBucketEventDataBlob =
+      userApi.events.fileSystem.NewBucket.is(newBucketEventEvent) && newBucketEventEvent.data;
+
+    if (!newBucketEventDataBlob) {
+      throw new Error("NewBucket event data does not match expected type");
+    }
+
+    const txs = [];
+    for (let i = 0; i < source.length; i++) {
+      const { fingerprint, file_size, location } =
+        await userApi.rpc.storagehubclient.loadFileInStorage(
+          source[i],
+          destination[i],
+          userApi.shConsts.NODE_INFOS.user.AddressId,
+          newBucketEventDataBlob.bucketId
+        );
+
+      txs.push(
+        userApi.tx.fileSystem.issueStorageRequest(
+          newBucketEventDataBlob.bucketId,
+          location,
+          fingerprint,
+          file_size,
+          userApi.shConsts.DUMMY_MSP_ID,
+          [userApi.shConsts.NODE_INFOS.user.expectedPeerId]
+        )
+      );
+    }
+
+    await userApi.sealBlock(txs, shUser);
+
+    // Allow time for the MSP to receive and store the file from the user
+    await sleep(3000);
+
+    const events = await userApi.assert.eventMany("fileSystem", "NewStorageRequest");
+
+    const matchedEvents = events.filter((e) =>
+      userApi.events.fileSystem.NewStorageRequest.is(e.event)
+    );
+
+    if (matchedEvents.length !== source.length) {
+      throw new Error(`Expected ${source.length} NewStorageRequest events`);
+    }
+
+    const issuedFileKeys = [];
+    for (const e of matchedEvents) {
+      const newStorageRequestDataBlob =
+        userApi.events.fileSystem.NewStorageRequest.is(e.event) && e.event.data;
+
+      if (!newStorageRequestDataBlob) {
+        throw new Error("Event doesn't match NewStorageRequest type");
+      }
+
+      const result = await mspApi.rpc.storagehubclient.isFileInFileStorage(
+        newStorageRequestDataBlob.fileKey
+      );
+
+      if (!result.isFileFound) {
+        throw new Error(
+          `File not found in storage for ${newStorageRequestDataBlob.location.toHuman()}`
+        );
+      }
+
+      issuedFileKeys.push(newStorageRequestDataBlob.fileKey);
+    }
+
+    // Seal block containing the MSP's transaction response to the storage request
+    await userApi.wait.mspResponseInTxPool();
+    await userApi.sealBlock();
+
+    let mspAcceptedStorageRequestDataBlob: any = undefined;
```

Comment:
whats going on with these empty try catch blocks and any data blobs?

if this fails in future it will be hard to debug and its not obvious what this test is trying to do.

I would instead have some branching logic that is determined by query of last block to see which of the conditions is fulfilled (and reject if none). 

Typescript doesn't have `match e { ... }` sadly, but we do have:

```ts
switch (true) {
   case e : {...} 
}
```

Instead of use the asserts to find the events like this, might more straight forward to search for the event manually by querying e.g. `(await api.query.system.events()).filter(x=> x.event.section.toString() === "fileSystem")`

and from there you can limit the amount of lets and untyped state hanging around  :cold_sweat:

Reply:
I have replaced it by searching manually. I also think it is better than `try and catch`